michael/src/reducers/gifReducer.js

```diff
@@ -0,0 +1,21 @@
+import initialState from './initialState';
+import {GET_GIF_LOADING, GET_GIF_SUCCESS, GET_GIF_FAIL} from '../actions/actionTypes';
+
+export default function gifReducer(state = initialState, action) {
+  let newState;
+  switch (action.type) {
+    case GET_GIF_LOADING:
+      console.log('loading')
+      return action;
+    case GET_GIF_SUCCESS:
+      newState = action;
+      console.log('success')
```

Comment:
As you go farther into the redux tutorial, this is what you'll want to change because when you get a successful request, you'll want to update the `newState` variable with the payload you get from the API. (this is meant to be a hint, so i wont' give you the full instructions on how to do this, haha)